src/validation/validatetx.cpp

```diff
@@ -0,0 +1,613 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "validatetx.h"
+
+#include "blockrelay/blockrelay_common.h"
+#include "blockstorage/blockstorage.h"
+#include "blockstorage/sequential_files.h"
+#include "checkpoints.h"
+#include "connmgr.h"
+#include "consensus/consensus.h"
+#include "consensus/merkle.h"
+#include "consensus/tx_verify.h"
+#include "dosman.h"
+#include "expedited.h"
+#include "init.h"
+#include "requestManager.h"
+#include "sync.h"
+#include "timedata.h"
+#include "txadmission.h"
+#include "txorphanpool.h"
+#include "ui_interface.h"
+#include "validationinterface.h"
+
+#include "txmempool.h"
+
+#include "validation.h"
+
+#include <boost/scope_exit.hpp>
+
+UniValue CheckInputsBetter(const CTransactionRef &ptx,
+    CValidationState &state,
```

Comment:
Seems like this is mostly a cut and paste from AcceptToMemoryPoolWorker?  I think it would be better not to have so much cut and paste code and make some reusable functions instead?

Reply:
or why not resuse AcceptToMemoryPool and just pass a flag indicating you only want validation not final mempool acceptance..and have it return whatever info you need.

Reply:
i chose not to do this because i changed the behavior of some of the code (removed asserts, continues instead of returns, etc.) as well as the building of the univalue object all the way through the functions. 
Although i never did testing, i think adding all of these checks would slow down the function/ node in general especially since it is one of the most called functions in the code base.

Reply:
I still think its going to be very hard to  maintain this code...I think it'll be very easy for us to forget to  migrate new changes to AcceptToMemoryPool.  I think that it would be better to not capture every single error and create this full univalue report at the end, but rather, just use ParallelAcceptoMemoryPool and capture any errors thrown in state from the return and print that out...sure the user may fix his tx and then resubmit only to find then next level error, but I think that would be a relatively rare case. IMO, this would solve the maintainability issue.